drivers/src/trng.rs

```diff
@@ -48,6 +57,48 @@ impl Trng {
         match self {
             Self::Internal(csrng) => Ok(csrng.generate12()?.into()),
             Self::External(trng_ext) => trng_ext.generate(),
+            Self::MfgMode(rng) => Ok(rng.generate12()?.into()),
```

Comment:
I don't think this match statement provides enough glitch resistance. If somebody can glitch the cpu into this branch when we're initializing the CFI PRNG, we lose significant CFI protections.

Reply:
Maybe that's ok if we add a bunch of cfi_asserts to FakeRng::generate()

Reply:
we probably need to look a lot closer at all this assembly.

Reply:
cfi_asserts are added to ROM main function to ensure the TRNG config correctness.